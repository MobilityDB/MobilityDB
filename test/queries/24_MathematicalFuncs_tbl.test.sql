/*****************************************************************************
 * Temporal addition
 *****************************************************************************/

SELECT count(*) FROM tbl_tint, tbl_int WHERE i + temp IS NOT NULL;
SELECT count(*) FROM tbl_tfloat, tbl_int WHERE i + temp IS NOT NULL;

SELECT count(*) FROM tbl_tint, tbl_float WHERE f + temp IS NOT NULL;
SELECT count(*) FROM tbl_tfloat, tbl_float WHERE f + temp IS NOT NULL;

SELECT count(*) FROM tbl_tint, tbl_int WHERE temp + i IS NOT NULL;
SELECT count(*) FROM tbl_tint, tbl_float WHERE temp + f IS NOT NULL;

SELECT count(*) FROM tbl_tint t1, tbl_tint t2 WHERE t1.temp + t2.temp IS NOT NULL;
SELECT count(*) FROM tbl_tint t1, tbl_tfloat t2 WHERE t1.temp + t2.temp IS NOT NULL;

SELECT count(*) FROM tbl_tfloat, tbl_int WHERE temp + i IS NOT NULL;
SELECT count(*) FROM tbl_tfloat, tbl_float WHERE temp + f IS NOT NULL;

SELECT count(*) FROM tbl_tfloat t1, tbl_tint t2 WHERE t1.temp + t2.temp IS NOT NULL;
SELECT count(*) FROM tbl_tfloat t1, tbl_tfloat t2 WHERE t1.temp + t2.temp IS NOT NULL;

/*****************************************************************************
 * Temporal subtraction
 *****************************************************************************/

SELECT count(*) FROM tbl_tint, tbl_int WHERE i - temp IS NOT NULL;
SELECT count(*) FROM tbl_tfloat, tbl_int WHERE i - temp IS NOT NULL;

SELECT count(*) FROM tbl_tint, tbl_float WHERE f - temp IS NOT NULL;
SELECT count(*) FROM tbl_tfloat, tbl_float WHERE f - temp IS NOT NULL;

SELECT count(*) FROM tbl_tint, tbl_int WHERE temp - i IS NOT NULL;
SELECT count(*) FROM tbl_tint, tbl_float WHERE temp - f IS NOT NULL;

SELECT count(*) FROM tbl_tint t1, tbl_tint t2 WHERE t1.temp - t2.temp IS NOT NULL;
SELECT count(*) FROM tbl_tint t1, tbl_tfloat t2 WHERE t1.temp - t2.temp IS NOT NULL;

SELECT count(*) FROM tbl_tfloat, tbl_int WHERE temp - i IS NOT NULL;
SELECT count(*) FROM tbl_tfloat, tbl_float WHERE temp - f IS NOT NULL;

SELECT count(*) FROM tbl_tfloat t1, tbl_tint t2 WHERE t1.temp - t2.temp IS NOT NULL;
SELECT count(*) FROM tbl_tfloat t1, tbl_tfloat t2 WHERE t1.temp - t2.temp IS NOT NULL;

/*****************************************************************************
 * Temporal multiplication
 *****************************************************************************/

SELECT count(*) FROM tbl_tint, tbl_int WHERE i * temp IS NOT NULL;
SELECT count(*) FROM tbl_tfloat, tbl_int WHERE i * temp IS NOT NULL;

SELECT count(*) FROM tbl_tint, tbl_float WHERE f * temp IS NOT NULL;
SELECT count(*) FROM tbl_tfloat, tbl_float WHERE f * temp IS NOT NULL;

SELECT count(*) FROM tbl_tint, tbl_int WHERE temp * i IS NOT NULL;
SELECT count(*) FROM tbl_tint, tbl_float WHERE temp * f IS NOT NULL;

SELECT count(*) FROM tbl_tint t1, tbl_tint t2 WHERE t1.temp * t2.temp IS NOT NULL;
SELECT count(*) FROM tbl_tint t1, tbl_tfloat t2 WHERE t1.temp * t2.temp IS NOT NULL;

SELECT count(*) FROM tbl_tfloat, tbl_int WHERE temp * i IS NOT NULL;
SELECT count(*) FROM tbl_tfloat, tbl_float WHERE temp * f IS NOT NULL;

SELECT count(*) FROM tbl_tfloat t1, tbl_tint t2 WHERE t1.temp * t2.temp IS NOT NULL;
SELECT count(*) FROM tbl_tfloat t1, tbl_tfloat t2 WHERE t1.temp * t2.temp IS NOT NULL;

/*****************************************************************************
 * Temporal division
 *****************************************************************************/

SELECT count(*) FROM tbl_tint, tbl_int WHERE i / temp IS NOT NULL;
SELECT count(*) FROM tbl_tfloat, tbl_int WHERE i / temp IS NOT NULL;

SELECT count(*) FROM tbl_tint, tbl_float WHERE f / temp IS NOT NULL;
SELECT count(*) FROM tbl_tfloat, tbl_float WHERE f / temp IS NOT NULL;

SELECT count(*) FROM tbl_tint, tbl_int WHERE temp / i IS NOT NULL;
SELECT count(*) FROM tbl_tint, tbl_float WHERE temp / f IS NOT NULL;

SELECT count(*) FROM tbl_tint t1, tbl_tint t2 WHERE t1.temp / t2.temp IS NOT NULL;
SELECT count(*) FROM tbl_tint t1, tbl_tfloat t2 WHERE t1.temp / t2.temp IS NOT NULL;

SELECT count(*) FROM tbl_tfloat, tbl_int WHERE temp / i IS NOT NULL;
SELECT count(*) FROM tbl_tfloat, tbl_float WHERE temp / f IS NOT NULL;

SELECT count(*) FROM tbl_tfloat t1, tbl_tint t2 WHERE t1.temp / t2.temp IS NOT NULL;
SELECT count(*) FROM tbl_tfloat t1, tbl_tfloat t2 WHERE t1.temp / t2.temp IS NOT NULL;

/*****************************************************************************
 * Temporal round and degrees
 *****************************************************************************/

SELECT count(*) FROM tbl_tfloat WHERE round(temp, 1) IS NOT NULL;
SELECT count(*) FROM tbl_tfloat WHERE degrees(temp) IS NOT NULL;

 /*****************************************************************************/

