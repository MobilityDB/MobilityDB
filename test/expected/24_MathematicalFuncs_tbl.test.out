/*****************************************************************************
 * Temporal addition
 *****************************************************************************/
SELECT count(*) FROM tbl_tint, tbl_int WHERE i + temp IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tfloat, tbl_int WHERE i + temp IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tint, tbl_float WHERE f + temp IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tfloat, tbl_float WHERE f + temp IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tint, tbl_int WHERE temp + i IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tint, tbl_float WHERE temp + f IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tint t1, tbl_tint t2 WHERE t1.temp + t2.temp IS NOT NULL;
 count 
-------
   154
(1 row)

SELECT count(*) FROM tbl_tint t1, tbl_tfloat t2 WHERE t1.temp + t2.temp IS NOT NULL;
 count 
-------
     4
(1 row)

SELECT count(*) FROM tbl_tfloat, tbl_int WHERE temp + i IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tfloat, tbl_float WHERE temp + f IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tfloat t1, tbl_tint t2 WHERE t1.temp + t2.temp IS NOT NULL;
 count 
-------
     4
(1 row)

SELECT count(*) FROM tbl_tfloat t1, tbl_tfloat t2 WHERE t1.temp + t2.temp IS NOT NULL;
 count 
-------
   150
(1 row)

/*****************************************************************************
 * Temporal subtraction
 *****************************************************************************/
SELECT count(*) FROM tbl_tint, tbl_int WHERE i - temp IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tfloat, tbl_int WHERE i - temp IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tint, tbl_float WHERE f - temp IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tfloat, tbl_float WHERE f - temp IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tint, tbl_int WHERE temp - i IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tint, tbl_float WHERE temp - f IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tint t1, tbl_tint t2 WHERE t1.temp - t2.temp IS NOT NULL;
 count 
-------
   154
(1 row)

SELECT count(*) FROM tbl_tint t1, tbl_tfloat t2 WHERE t1.temp - t2.temp IS NOT NULL;
 count 
-------
     4
(1 row)

SELECT count(*) FROM tbl_tfloat, tbl_int WHERE temp - i IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tfloat, tbl_float WHERE temp - f IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tfloat t1, tbl_tint t2 WHERE t1.temp - t2.temp IS NOT NULL;
 count 
-------
     4
(1 row)

SELECT count(*) FROM tbl_tfloat t1, tbl_tfloat t2 WHERE t1.temp - t2.temp IS NOT NULL;
 count 
-------
   150
(1 row)

/*****************************************************************************
 * Temporal multiplication
 *****************************************************************************/
SELECT count(*) FROM tbl_tint, tbl_int WHERE i * temp IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tfloat, tbl_int WHERE i * temp IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tint, tbl_float WHERE f * temp IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tfloat, tbl_float WHERE f * temp IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tint, tbl_int WHERE temp * i IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tint, tbl_float WHERE temp * f IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tint t1, tbl_tint t2 WHERE t1.temp * t2.temp IS NOT NULL;
 count 
-------
   154
(1 row)

SELECT count(*) FROM tbl_tint t1, tbl_tfloat t2 WHERE t1.temp * t2.temp IS NOT NULL;
 count 
-------
     4
(1 row)

SELECT count(*) FROM tbl_tfloat, tbl_int WHERE temp * i IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tfloat, tbl_float WHERE temp * f IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tfloat t1, tbl_tint t2 WHERE t1.temp * t2.temp IS NOT NULL;
 count 
-------
     4
(1 row)

SELECT count(*) FROM tbl_tfloat t1, tbl_tfloat t2 WHERE t1.temp * t2.temp IS NOT NULL;
 count 
-------
   150
(1 row)

/*****************************************************************************
 * Temporal division
 *****************************************************************************/
SELECT count(*) FROM tbl_tint, tbl_int WHERE i / temp IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tfloat, tbl_int WHERE i / temp IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tint, tbl_float WHERE f / temp IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tfloat, tbl_float WHERE f / temp IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tint, tbl_int WHERE temp / i IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tint, tbl_float WHERE temp / f IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tint t1, tbl_tint t2 WHERE t1.temp / t2.temp IS NOT NULL;
 count 
-------
   154
(1 row)

SELECT count(*) FROM tbl_tint t1, tbl_tfloat t2 WHERE t1.temp / t2.temp IS NOT NULL;
 count 
-------
     4
(1 row)

SELECT count(*) FROM tbl_tfloat, tbl_int WHERE temp / i IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tfloat, tbl_float WHERE temp / f IS NOT NULL;
 count 
-------
 10000
(1 row)

SELECT count(*) FROM tbl_tfloat t1, tbl_tint t2 WHERE t1.temp / t2.temp IS NOT NULL;
 count 
-------
     4
(1 row)

SELECT count(*) FROM tbl_tfloat t1, tbl_tfloat t2 WHERE t1.temp / t2.temp IS NOT NULL;
 count 
-------
   150
(1 row)

/*****************************************************************************
 * Temporal round and degrees
 *****************************************************************************/
SELECT count(*) FROM tbl_tfloat WHERE round(temp, 1) IS NOT NULL;
 count 
-------
   100
(1 row)

SELECT count(*) FROM tbl_tfloat WHERE degrees(temp) IS NOT NULL;
 count 
-------
   100
(1 row)

/*****************************************************************************/
